lib/dry/cli/utils/files/file_system.rb

```diff
@@ -0,0 +1,190 @@
+# frozen_string_literal: true
+
+require "fileutils"
+
+module Dry
+  class CLI
+    module Utils
+      class Files
+        # File System abstraction to support `Dry::CLI::Utils::Files`
+        #
+        # @since x.x.x
+        # @api private
+        class FileSystem
+          # Creates a new instance
+          #
+          # @param file [Class]
+          # @param file_utils [Class]
+          #
+          # @return [Dry::CLI::Utils::Files::FileSystem]
+          #
+          # @since x.x.x
+          def initialize(file: File, file_utils: FileUtils)
```

Comment:
I'm wondering if this is going to be leveraged in any way? Seems like this class could be just a simple wrapper around File and FileUtils + a couple of additional methods.

Reply:
@solnic I didn't get you here. Do you mean that:

  1. `FileSystem` doesn't make sense to exist?
  2. Nobody will leverage DI in `#initialize` and so we can hardcode deps?
  3. `FileSystem` should expose class method interface?

Reply:
@jodosha no, it makes a lot of sense to have this class - but I did mean that maybe it doesn't make sense to inject `File` and `FileUtils` and just use them internally as an implementation detail w/o providing the ability to override them. It could become a class-only interface too, actually.